### PR TITLE
Registration page validation

### PR DESCRIPTION
**Pull request by profileEnumerable**
Pay attention that new custom pipe(firstError) was added 

This pipe intended for:
-  returning name of the first validation error from FormControl.errors object
-  avoiding overlapping validation messages when two validation rules was violated simultaneously. See screenshot bellow
image

### CODE REVIEW

File: Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html [hunk 1]
```diff
@@ -16,33 +16,41 @@ <h1 class="registration__title">Getting started with <span>Campus</span></h1>
...
+                  <small *ngSwitchCase="'required'">
                   Last name is required
                 </small>
-                <small *ngIf="registerForm.get('last_name').errors.pattern">
-                  Only english letters
+                  <small *ngSwitchCase="'maxlength'">
+                    Last name > 50 symbols
                 </small>
+                  <small *ngSwitchCase="'pattern'">
+                    Only English letters
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html:48]: It's better to use message **Only Latin characters are allowed**

File: Src/Campus.Master.API/UI/src/app/pipes/first-error.pipe.spec.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { FirstErrorPipe } from './first-error.pipe';
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/pipes/first-error.pipe.spec.ts:1]: Please add more functionality tests :slightly_smiling_face:

File: Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html [hunk 3]
```diff
@@ -16,33 +16,41 @@ <h1 class="registration__title">Getting started with <span>Campus</span></h1>
...
+                  <small *ngSwitchCase="'required'">
                   First name is required
                 </small>
-                <small *ngIf="registerForm.get('first_name').errors.pattern">
-                  Only english letters
+                  <small *ngSwitchCase="'maxlength'">
+                    First name > 50 symbols
                 </small>
+                  <small *ngSwitchCase="'pattern'">
+                    Only English letters
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html:29]: It's better to use message **Only Latin characters are allowed**

File: Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.ts [hunk 4]
```diff
@@ -37,24 +37,27 @@ export class RegistrationPageComponent implements OnInit, OnDestroy {
...
-          Validators.pattern('[a-zA-Z]*'),
+          Validators.maxLength(50),
+          Validators.pattern('[a-zA-Z]*')
         ]),
         email: new FormControl(null, [Validators.required, Validators.email]),
         password: new FormControl(null, [
           Validators.required,
           Validators.minLength(8),
+          Validators.maxLength(100),
           Validators.pattern('(?=.*?[0-9])(?=.*?[A-Z]).{8,}.+')
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.ts:53]: This validation case doesn't work according to the use cases. Need to be fixed.

**Review by antonAce — COMMENTED** [review]
Looks promising, just fix these comments

File: Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html (hunk 1 above)
Comment (profileEnumerable) [reply Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html]: Replaced with more short message, due to small gap between messages in Fist/LastName fields

File: Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html (hunk 3 above)
Comment (profileEnumerable) [reply Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html]: Replaced with more short message, due to small gap between messages in Fist/LastName fields

Comment (antonAce) [reply Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html]: If so, you can replace the message with smth like this one **Only Latin**

File: Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html (hunk 1 above)
Comment (antonAce) [reply Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.html]: If so, you can replace the message with smth like this one **Only Latin**

File: Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.ts (hunk 4 above)
Comment (profileEnumerable) [reply Src/Campus.Master.API/UI/src/app/registration-page/registration-page.component.ts]: Fixed

File: Src/Campus.Master.API/UI/src/app/pipes/first-error.pipe.spec.ts (hunk 2 above)
Comment (profileEnumerable) [reply Src/Campus.Master.API/UI/src/app/pipes/first-error.pipe.spec.ts:1]: Added

**Review by antonAce — APPROVED** [review]
Well done :grinning:

**Pull request merged**